test/features/perform_sanity_checks_on_reviews_test.rb

```diff
@@ -0,0 +1,166 @@
+      assert has_content? 'Granule G309210-GHRC/6 has been successfully associated to this collection revision 8'
+
+      # Now lets add a review comment
+      see_collection_revision_details(8)
+      assign_review_comments('my review comment', 'my report comment')
+
+      # We should see some informative issues.  These won't prevent saving the review.
+      assert has_content? 'Note, not all columns have been flagged with a color, be sure this is done before marking this review complete.'
+      assert has_content? 'Note, some columns still need a second opinion review.'
+    end
+
+    # in this case short name, long name are marked with a color, but version id is not in the granule
+    it 'checks granule review during mark as done and produces error' do
+      # Now click MARK AS DONE
+      click_button 'MARK AS DONE'
+      assert_equal Record.find_by(id: 1).state, 'in_arc_review'
+      assert_equal Record.find_by(id: 1).state, Record.find_by(id: 16).state
+
+      # We should see errors and will prevent the collection from moving to ready for daac review.
+      assert has_content?('Not all columns in the associated granule have been flagged with a color!')
+      assert has_content?('The associated granule needs two completed reviews')
+      assert has_content? 'Record failed to update.'
+      assert_no_css '#done_button[disabled]' # still enabled
+    end
+
+    it 'fixes granule and then checks collection and produces error' do
+      # Lets back out to fix the granule issues.
+      find('#nav_back_button').click
+
+      see_granule_revision_details(6)
+      assign_review_comments('my review comment', 'my report comment')
+
+      # Now lets fix issue #1 and make sure the column has a flagged color
+      click_button 'Collection Info'
+
+      find('#color_button_green_ShortName').click
+      find('.review-toggle__text').click
+      #
+      # Lets save the review and back out again.
+      click_button 'Save Review'
+
+      find('#nav_back_button').click
+      find('#nav_back_button').click
+
+      # Lets view the collection review details again
+      see_collection_revision_details(8)
+
+      # Now click mark as done should reveal a new error, this time with the collection.
+      click_button 'MARK AS DONE'
+
+      assert_equal Record.find_by(id: 1).state, 'in_arc_review'
+      assert_equal Record.find_by(id: 1).state, Record.find_by(id: 16).state
+
+      assert has_no_content? 'Not all columns in the associated granule have been flagged with a color!'
+      assert has_no_content? 'The associated granule needs two completed reviews'
+      assert has_content? 'Not all columns have been flagged with a color, cannot close review.'
+      assert has_content? 'Record failed to update.'
+    end
+
+    it 'fixes granule, fixes collection and produces success message -- should ignore second opinion checks, then fixes second opinions checks' do
```

Comment:
This test is testing three different bits of related functionality.  In the MMT, we would typically do this with nested context blocks.  
```
context
  before do
    the stuff
  end

  it do
    test stuff
  end

  context
    before do
      some more stuff
    end
    ...
```
And I think it would make this test more maintainable.